### PR TITLE
Limit tags length

### LINKED ISSUE #175: Display problems with meme
If there are too many tags they will push the other content downwards. See screenshot.

When description is too long you can also see that it is truncated and not visible.

<hr><sub>[original: crazoter/ped#9]<br/>
</sub>

### LINKED ISSUE #243: Tag length issue
Limit length of tag for display, or limit the length of tag in general (since user won't need such a long tag.)
image

### PR DESCRIPTION
**Pull request by moziliar**
closes #243 Tag length issue
Limit length of tag name to 38 characters (about the same width as a MemeCard). Limit number of tags display in MemeCard in Memes panel. If there are undisplayed tags, display ".." as a placeholder. This is to prevent tag overflow.
Closes #175

### CODE REVIEW

File: src/main/java/seedu/weme/ui/MemeCard.java [hunk 1]
```diff
@@ -85,6 +108,40 @@ public void update(Meme meme, int newIndex, SimpleIntegerProperty numOfLikes, Si
                 dislikes.setText(Integer.toString((int) newValue)));
     }
 
+    private int getTagLimit(Meme meme) {
```
Comment (jonchan51) [inline src/main/java/seedu/weme/ui/MemeCard.java]: Explain what this method does pls, or at least the reasoning behind it

File: src/main/java/seedu/weme/ui/MemeCard.java [hunk 2]
```diff
@@ -74,9 +92,14 @@ public void update(Meme meme, int newIndex, SimpleIntegerProperty numOfLikes, Si
         id.setText(newIndex + "");
         description.setText(meme.getDescription().value);
         tags.getChildren().clear();
+        int limit = getTagLimit(meme);
         meme.getTags().stream()
                 .sorted(Comparator.comparing(tag -> tag.tagName))
+                .limit(limit)
                 .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
+        if (limit < meme.getTags().size()) {
+            tags.getChildren().add(new Label(TAG_TRUNCATE_TEXT));
```
Comment (jonchan51) [inline src/main/java/seedu/weme/ui/MemeCard.java]: any way to make this appear like '..' without looking like a label?

File: src/main/java/seedu/weme/model/tag/Tag.java [hunk 3]
```diff
@@ -9,8 +9,9 @@
  */
 public class Tag {
 
-    public static final String MESSAGE_CONSTRAINTS = "Tags names should be alphanumeric";
+    public static final String MESSAGE_CONSTRAINTS = "Tags names should be alphanumeric and at most 35 characters.";
     public static final String VALIDATION_REGEX = "\\p{Alnum}+";
+    public static final int TAG_MAX_CHAR = 38;
```
[the remark this answers is not in the document]
Comment (moziliar) [reply src/main/java/seedu/weme/model/tag/Tag.java]: 38 would work, unless you want 35 instead.

File: src/main/java/seedu/weme/ui/MemeCard.java (hunk 1 above)
Comment (moziliar) [reply src/main/java/seedu/weme/ui/MemeCard.java]: this method gets the max number of tags to be displayed s.t. the content won't overflow.

**Pull request merged**